File: shared-atlas-dart/lib/src/interfaces/isigned_url_response.dart
class ISignedUrlResponse {
  final String url;
  final String path;
  final num expiresAt;
  final num expiresIn;

  ISignedUrlResponse({
    required this.url,
    required this.path,
    required this.expiresAt,
    required this.expiresIn,
  });

  factory ISignedUrlResponse.fromJson(Map<String, dynamic> json) {
    return ISignedUrlResponse(
      url: json['url'] as String,
      path: json['path'] as String,
      expiresAt: json['expiresAt'] as num,
      expiresIn: json['expiresIn'] as num,
    );
  }
  Map<String, dynamic> toJson() {
    return {
      'url': this.url,
      'path': this.path,
      'expiresAt': this.expiresAt,
      'expiresIn': this.expiresIn,
    };
  }
}

File: shared-atlas-rust/src/interfaces/isigned_url_response.rs
use serde::{Serialize, Deserialize};

/// Signed URL Response
/// 签名 URL 响应
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ISignedUrlResponse {
/// New signed URL
/// 新签名 URL
    pub url: String,
/// GCS path (for verification)
/// GCS 路径（用于验证）
    pub path: String,
/// URL expiration timestamp (Unix milliseconds)
/// URL 过期时间戳（Unix 毫秒）
    pub expires_at: f64,
/// Seconds until expiration
/// 距离过期的秒数
    pub expires_in: f64,
}